controller/controller_server/csi_controller_server.py

```diff
@@ -418,7 +419,10 @@ def CreateSnapshot(self, request, context):
 
                 volume_name = array_mediator.get_volume_name(vol_id)
                 logger.info("Snapshot name : {}. Volume name : {}".format(snapshot_name, volume_name))
-                snapshot = array_mediator.get_snapshot(snapshot_name)
+                snapshot = array_mediator.get_snapshot(
+                    snapshot_name,
+                    volume_context=request.parameters
```

Comment:
I think we should avoid passing `volume_context` altogether, and pass `pool` instead (including existing code, e.g. `create_volume`)